lib/nerves_hub_link/extensions/health/metric_set.ex

```diff
@@ -0,0 +1,7 @@
+defmodule NervesHubLink.Extensions.Health.MetricSet do
+  @moduledoc """
+  Behaviour for implementing a custom metric set to be used in a report.
+  """
+
+  @callback metrics() :: %{String.t() => number()}
```

Comment:
```suggestion
  @callback metrics() :: %{String.t() | atom() => number()}
```
This says `String.t()` keys but several of the default metric_set implementations are atom keys. Allowing both would probably work for simplicity, but merging later on needs to be explicit